### PR TITLE
DietPi-Software | Simplify and consistent config file naming

### PR DESCRIPTION
**Pull request by MichaIng**
**Status**: Ready
- [x] Pydio
- [x] Nextcloud
- [x] ownCloud
- [x] dietpi-software
- [x] dietpi-letsencrypt
- [x] Search code
- [x] Patching
- [x] Changelog

**Commit list/description**:
- nginx.pydio.conf
- apache.nextcloud.conf
- nginx.nextcloud.conf
- 99-dietpi-kodi.rules
... all of them 😄

### CODE REVIEW

**Review by MichaIng — COMMENTED** [review]
@Fourdee 
Ah the idea was to just use `conf` here for config files?
Shall we then do this for other software titles as well?

Only concern I have is, if e.g. we remove a software title from DietPi and replace the ID, the old dietpi-software versions will not recognize and place a wrong config file.
If those have a more specific name, then at least the user will face a meaningful error, can upgrade, ask or inform himself about dropped software.

**Comment by MichaIng** [comment]
@Fourdee 
`.conf/dps_6/50-rk3399.rules` is not yet in use, should this be placed on Xserver install on this device?

**Comment by MichaIng** [comment]
Damn... /etc/nginx/conf.d is main config level while /etc/nginx/sites-dietpi was server config level (included by sites-enabled/default).
Need to revert, no other solution than having an own include dir 🤔.
- [x] Done

**Comment by Fourdee** [comment]
@MichaIng 

> Only concern I have is, if e.g. we remove a software title from DietPi and replace the ID, the old dietpi-software versions will not recognize and place a wrong config file.

Should not matter, as master branch will not change until we PR. Just need to make sure we update ```dietpi-software``` as needed, and, if we remove/swap software, set installed state to 0 on users system.

**Comment by Fourdee** [comment]
@MichaIng 

> .conf/dps_6/50-rk3399.rules is not yet in use, should this be placed on Xserver install on this device?

Ignore it for now, testing/getting it ready only.

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -7748,16 +7745,15 @@ url.redirect = (
...
 
-					#???Enable rewrite mod
 					echo '# Redirect Cal/CardDAV requests to ownCloud endpoint:
 location = /.well-known/carddav {
 	return 301 $scheme://$host/owncloud/remote.php/dav;
 }
 location = /.well-known/caldav {
 	return 301 $scheme://$host/owncloud/remote.php/dav;
-}' > /etc/nginx/sites-dietpi/dietpi-dav_redirect.config
+}' > /etc/nginx/sites-dietpi/dietpi-dav_redirect.conf
```
Comment (Fourdee) [inline dietpi/dietpi-software:7756]: Needs verifying if Nginx supports the ```.conf``` extension. Might require ```.config```?

Comment (Fourdee) [reply dietpi/dietpi-software:7756]: See it passed local testing, will merge and test also:
https://github.com/Fourdee/DietPi/issues/2308#issuecomment-443547842

**Comment by Fourdee** [comment]
@MichaIng 

Thanks Micha, looks good, will merge 👍

File: dietpi/dietpi-software (hunk 1 above)
Comment (MichaIng) [reply dietpi/dietpi-software:7756]: @Fourdee
Jep file ending can be freely chosen by include directive. Also inside conf.d it is .conf, similar to all other webservers. So it's more constant to stay with this.

**Pull request merged**